src/graphics/resources/fibers.rs

```diff
@@ -25,12 +27,23 @@ impl FiberResources {
     }
 }
 
-fn fiber_geometry(fibers: Reader) -> (Vec<FiberVertex>, Vec<u32>) {
-    let (mut vertices, ranges) = fiber_vertices(fibers);
+pub fn batches(fibers: Reader, client: &Client) -> Vec<FiberBatch> {
+    let mut iter = fibers.into_streamlines_iter().peekable();
+    let mut batches = vec![];
+
+    while iter.peek().is_some() {
+        let streamlines = iter.by_ref().take(client.streamline_batch_size).collect();
+        batches.push(FiberBatch::new(streamlines, &client.device));
+    }
```

Comment:
I'm surprised but the way you wrote this loop. Can you please explain in your own words why you used the "peek" concept instead of a standard iteration?

Reply:
There are probably room for improvement here. Here's how it works:

There are two iterations at play here, the first one looks for the end and the other one retrieve a batch of streamlines (`.take(streamline_batch_size)`). Taking a batch mutates the iterator. A basic iterator can only be queried with `next()` which is why I used a peekable iterator, it allows us to look if the iterator is at the end without modifing it.

I'm not sure how I would be able to implement this logic with a regular iterator. Let me know if there's a better way or if my explanation is unclear.

Reply:
Does something like [this](https://users.rust-lang.org/t/how-to-breakup-an-iterator-into-chunks/87915/5) solve your problem? Maybe it's uselessly complex. I'm not sure. I fell like there's a cost to use peekable, but I'm probably wrong.

Reply:
Yes, this works. This is basically `chunks(..)` but for an iterator instead of a `Vec`. I don't think the overhead is too bad, this iteration runs 26 times on the whole brain tracking using the default batch size.

Here's the new version:
```rust
pub fn batches(fibers: Reader, client: &Client) -> Vec<FiberBatch> {
    let mut iter = fibers.into_streamlines_iter();

    std::iter::from_fn(|| {
        let streamlines: Vec<Streamline> =
            iter.by_ref().take(client.streamline_batch_size).collect();

        (!streamlines.is_empty()).then(|| FiberBatch::new(streamlines, &client.device))
    })
    .collect()
}
```